dual_net.py

```diff
@@ -321,58 +193,133 @@ def my_res_layer(inputs, train_mode):
         policy_output * tf.log(policy_output), axis=1))
-    boundaries = list(map(int, [1e6, 2 * 1e6]))
+    boundaries = [int(1e6), int(2e6)]
     values = [1e-2, 1e-3, 1e-4]
     learning_rate = tf.train.piecewise_constant(
         global_step, boundaries, values)
     update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
     with tf.control_dependencies(update_ops):
         train_op = tf.train.MomentumOptimizer(
-            learning_rate, hparams['momentum']).minimize(
+            learning_rate, params['momentum']).minimize(
                 combined_cost, global_step=global_step)
 
-    return {
-        'policy_cost': policy_cost,
-        'value_cost': value_cost,
-        'l2_cost': l2_cost,
-        'combined_cost': combined_cost,
-        'policy_entropy': policy_entropy,
-        'global_step': global_step,
-        'train_op': train_op,
-    }
-
 
-def logging_ops():
-    return tf.summary.merge([
-        tf.summary.histogram(weight_var.name, weight_var)
-        for weight_var in tf.trainable_variables()],
-        name="weight_summaries")
+    metric_ops = {
+        'accuracy': tf.metrics.accuracy(labels=labels['pi_tensor'],
+                                        predictions=policy_output,
+                                        name='accuracy_op'),
+        'policy_cost': tf.metrics.mean(policy_cost),
+        'value_cost': tf.metrics.mean(value_cost),
+        'l2_cost': tf.metrics.mean(l2_cost),
+        'policy_entropy': tf.metrics.mean(policy_entropy),
+        'combined_cost': tf.metrics.mean(combined_cost),
+    }
+    # Register metrics as tf.summary to make them show up during training
+    for metric_name, metric_op in metric_ops.items():
+        tf.summary.scalar(metric_name, metric_op[1])
+
+    return tf.estimator.EstimatorSpec(
+        mode=mode,
+        predictions={
+            'policy_output': policy_output,
+            'value_output': value_output,
+        },
+        loss=combined_cost,
+        train_op=train_op,
+        eval_metric_ops=metric_ops,
+        )
+
+
+def get_estimator(working_dir, **hparams):
+    hparams = get_default_hyperparams(**hparams)
+    return tf.estimator.Estimator(
+        model_fn,
+        model_dir=working_dir,
+        params=hparams)
+
+
+def bootstrap(working_dir, **hparams):
+    """Initialize a tf.Estimator run with random initial weights.
+
+    Args:
+        working_dir: The directory where tf.estimator will drop logs,
+            checkpoints, and so on
+        hparams: hyperparams of the model.
+    """
+    hparams = get_default_hyperparams(**hparams)
+    # a bit hacky - forge an initial checkpoint with the name that subsequent
+    # Estimator runs will expect to find.
+    #
+    # Estimator will do this automatically when you call train(), but calling
+    # train() requires data, and I didn't feel like creating training data in
+    # order to run the full train pipeline for 1 step.
+    estimator_initial_checkpoint_name = 'model.ckpt-1'
```

Comment:
what happens if this file exists?

Reply:
Hm. Would probably get overwritten. Probably not a big deal, because old checkpoints are not really used other than for backup purposes.